Replace Grunt with Webpack to compile assets

rubenmoya opened the pull request:
### Description
This PR aims to replace our current Grunt tasks for compiling our assets to Webpack to simplify our building process. It also removes unnecessary fonts and images from editor. And updates Webpack to version 4.

### Pages migrated
**Dashboard**
✅ Account
✅ Dashboard
✅ Data Library
✅ API Keys
✅ Mobile Apps
✅ Organization
✅ Profile
✅ User feed (public)
✅ Public maps (public)
✅ Public datasets (public)
✅ Public table (public)
✅ Log in
✅ Sign up
✅ Confirmation

**Builder**
✅ Dataset
✅ Builder
✅ Embed
✅ Embed protected


### Assets migrated
- CSS for the pages listed above
- `/assets/images/` folder copied to `[version]/images`
- `/public/favicons` folder copied to `[version]/favicons`
- CartoAssets fonts are extracted to `[version]/fonts`
- `['avatars/*.png', 'alphamarker.png', 'google-maps-basemap-icons/*.jpg', 'carto.png']` copied to `/unversioned/images`

### Acceptance steps
- Deploy to a new staging instance
  - [ ] Check the assets are generated correctly
  - [ ] Check every page listed above looks good
  - [ ] Check the pages marked as public work both, with builder and with editor


### Build times
This only includes `dashboard`, `builder` and `css` builds, we're not building `editor`.

**Before:**
![screen shot 2018-06-15 at 08 49 34](https://user-images.githubusercontent.com/905225/41454476-cb4b3f30-7079-11e8-8b17-069c19a1d452.png)
![screen shot 2018-06-15 at 08 52 15](https://user-images.githubusercontent.com/905225/41454477-cb64f18c-7079-11e8-8614-becd239edaa2.png)
![screen shot 2018-06-15 at 08 54 15](https://user-images.githubusercontent.com/905225/41454478-cb7e6bbc-7079-11e8-960d-42938c5d39e6.png)
Total: ~7:45 minutes 

**After:**
![screen shot 2018-06-15 at 09 30 55](https://user-images.githubusercontent.com/905225/41455859-eba415b8-707e-11e8-8f2b-3606a3e7c02d.png)
Time: 2:10 minutes

So... that's a decrease of 72% in our building time :) also, before we had to run 2 webpack instances and a grunt one to watch the file changes, that was leading to some problems with processes not being closed so we had to kill the processes manually, we shouldn't have that problem now.

rubenmoya commented:
retest this please

rubenmoya commented:
retest this please

ivanmalagon commented:
Spotted on an acceptance on the webpack branch so far.

**Changing dataset privacy on dataset page**

z-index of paginator is broken.

![image](https://user-images.githubusercontent.com/1078228/41471956-4fc19850-70b5-11e8-9e26-54e4b181d4e5.png)

**Public map view of a user with 1 map**

Are the white squares what it's expected to see here?

![image](https://user-images.githubusercontent.com/1078228/41472163-ec57902a-70b5-11e8-9c3d-80cefede497f.png)

ivanmalagon commented:
Yes, white squares are shown now in production. Not an issue then.

rubenmoya commented:
Ramiro found this while doing the acceptance:

![image](https://user-images.githubusercontent.com/905225/41536049-b1c4b158-7304-11e8-9315-8b57f95935b5.png)

I looks like I removed the `user_feed` css import somehow when I added the conditional for builder/editor. It's fixed now.

ivanmalagon commented:
We've added a config property in `app_config.yml` to be able to bypass the static pages

`bypass_static_pages: true`

Related to https://github.com/CartoDB/cartodb/issues/14038

I'm sure we'll override this with a better mechanism in the linked issue.

ivanmalagon commented on webpack/v4/webpack.base.config.js:
```diff
@@ -0,0 +1,245 @@
+      },
+      {
+        test: /\.js$/,
+        loader: 'babel-loader',
```
We were suppossed to transpile also Builder, right? I'm missing that `rootDir`.

rubenmoya replied:
Seems like we lost it somehow in the merge, because the commit it's up there

ivanmalagon commented on webpack/v4/webpack.base.config.js:
```diff
@@ -0,0 +1,245 @@
+          rootDir('lib/assets/javascripts/dashboard')
+        ],
+        options: {
+          presets: ['env'],
```
Does this transpile for IE11?

rubenmoya replied:
It's the same one we were using for the dashboard, so I'm guessing it does
